src/doc/rustc-dev-guide/src/tests/ui.md

```diff
@@ -335,6 +335,9 @@ But for strict testing, try to use the `ERROR` annotation as much as possible,
 including `//~?` annotations for diagnostics without span.
 For compile time diagnostics `error-pattern` should very rarely be necessary.
 
+Per-line annotations (`//~`) are still checked in tests using `error-pattern`,
+to opt out of these checks in exceptional cases use `//@ compile-flags: --error-format=human`.
```

Comment:
Discussion: hm, that feels somewhat hacky (and difficult to infer from reading the test). Should we instead introduce sth more explicit like `//@ dont-check-per-line-annotations` (long, but its usage should be quite rare that this shouldn't be a problem)?

Reply:
I think I'd rather use an existing option if it works, we already sort of have a proliferation of exotic options (#134888).

`--error-format=human` says "don't check the json, only the rendered text", which seems appropriate in cases like this. A new option is unlikely to be more discoverable.
(Perhaps we can reconsider when I migrate all of the remaining `error-pattern`s.)

Reply:
Seems fair!